frontend/src/pages/dashboard/profile/EditProfile.tsx

```diff
@@ -65,30 +65,51 @@ export default function EditProfileForm() {
     sncsCertificationId: Yup.number().optional(),
   });
 
-  const emptyFile = {} as File;
-
-  const defaultValues = {
-    ...currentUser,
-    avatar: {
-      ...emptyFile,
-      preview: currentUser?.profilePictureUrl || '',
-    },
-    favouriteGymIds: currentUser?.favouriteGyms?.map((gym) => gym.id),
-  };
+  // const emptyFile = useMemo(() => ({} as File), []);
+
+  const defaultValues = useMemo(
+    () => ({
+      name: currentUser.name ? currentUser.name : undefined,
+      height: currentUser.height ? currentUser.height : undefined,
+      reach: currentUser.reach ? currentUser.reach : undefined,
+      pronounId: currentUser.pronounId ? currentUser.pronounId : undefined,
+      highestBoulderingGradeId: currentUser.highestBoulderingGradeId
+        ? currentUser.highestBoulderingGradeId
+        : undefined,
+      highestTopRopeGradeId: currentUser.highestTopRopeGradeId
+        ? currentUser.highestTopRopeGradeId
+        : undefined,
+      highestLeadClimbingGradeId: currentUser.highestLeadClimbingGradeId
+        ? currentUser.highestLeadClimbingGradeId
+        : undefined,
+      favouriteGymIds: currentUser.favouriteGyms
+        ? currentUser.favouriteGyms?.map((gym) => gym.id)
+        : undefined,
+      sncsCertificationId: currentUser.sncsCertificationId
+        ? currentUser.sncsCertificationId
+        : undefined,
```

Comment:
React Hook Form does not like undefined values; this will result in UI bugs where the input label overlays on top of the input value. Hence, setting undefined as an input value is not recommended.

Currently, if user data has not loaded in yet, useGetIdentity returns a defaultUser where the string fields are blank strings and numeric fields are 0. This avoids the bug mentioned above.

Hence, I think there is no need to conditionally set the defaultValue based on currentUser, as this logic is already handled in useGetRequest. (Unless there was a bug with the previous implementation)